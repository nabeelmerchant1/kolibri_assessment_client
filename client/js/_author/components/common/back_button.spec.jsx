import React        from 'react';
import { shallow }  from 'enzyme';
import BackButton   from './back_button';

describe('back_button component', () => {
  let props;
  let result;
  let calledFunc;

  beforeEach(() => {
    calledFunc = false;
    props = {
      handleClick: () => { calledFunc = true; },
    };
    result = shallow(<BackButton {...props} />);
  });

  it('matches the snapshot taken', () => {
    expect(result).toMatchSnapshot();
  });

  it('calls the handleClick function onClick', () => {
    expect(calledFunc).toBeFalsy();
    result.find('button').simulate('click');
    expect(calledFunc).toBeTruthy();
  });
});
