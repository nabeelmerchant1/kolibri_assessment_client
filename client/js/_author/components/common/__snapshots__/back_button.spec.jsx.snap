// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`back_button component matches the snapshot taken 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <button
    className="au-c-btn au-c-btn--sm au-c-btn--outline au-c-btn--back"
    onClick={[Function]}
>
    <i
        className="material-icons"
    >
        keyboard_arrow_left
    </i>
    Back
</button>,
  "nodes": Array [
    <button
      className="au-c-btn au-c-btn--sm au-c-btn--outline au-c-btn--back"
      onClick={[Function]}
>
      <i
            className="material-icons"
      >
            keyboard_arrow_left
      </i>
      Back
</button>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <backButton
        handleClick={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleClick": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <button
          className="au-c-btn au-c-btn--sm au-c-btn--outline au-c-btn--back"
          onClick={[Function]}
>
          <i
                    className="material-icons"
          >
                    keyboard_arrow_left
          </i>
          Back
</button>,
        "_debugID": 2,
        "_renderedOutput": <button
          className="au-c-btn au-c-btn--sm au-c-btn--outline au-c-btn--back"
          onClick={[Function]}
>
          <i
                    className="material-icons"
          >
                    keyboard_arrow_left
          </i>
          Back
</button>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <backButton
    handleClick={[Function]}
/>,
}
`;
